packages/aws-cdk-lib/custom-resources/lib/custom-resource-config/custom-resource-config.ts

```diff
@@ -0,0 +1,79 @@
+import { IConstruct, MetadataEntry } from 'constructs';
+import * as cloudformation from '../../../aws-cloudformation';
+import * as lambda from '../../../aws-lambda';
+import * as logs from '../../../aws-logs';
+import { IAspect, Aspects } from '../../../core/lib';
+
+export const CUSTOM_RESOURCE_PROVIDER = 'aws:cdk:is-custom-resource-handler-customResourceProvider';
+export const CUSTOM_RESOURCE_SINGLETON = 'aws:cdk:is-custom-resource-handler-singleton';
+export const CUSTOM_RESOURCE_SINGLETON_LOG_GROUP = 'aws:cdk:is-custom-resource-handler-logGroup';
+export const CUSTOM_RESOURCE_SINGLETON_LOG_RETENTION = 'aws:cdk:is-custom-resource-handler-logRetention';
+
+/**
+ * Manages AWS vended Custom Resources
+ */
+export class CustomResourceConfig {
+  /**
+   * Returns the tags API for this scope.
+   * @param scope The scope
+   */
+  public static of(scope: IConstruct): CustomResourceConfig {
+    return new CustomResourceConfig(scope);
```

Comment:
a user shouldn't be able to add 2 of these on the same scope. Can you see if it's feasible to throw an error if they've already specified a `CustomResourceConfig` on this scope? `Aspects.of(this.scope).all` should give you all the existing aspects.